### PR TITLE
Gradient generation tool to support GENN models in FOQUS

### PR DESCRIPTION
**Pull request by bpaul4**
## Summary/Motivation:

Gradient generation tool from Brady Gess. Certain machine learning tools train gradient-enhanced neural network (GENN) models which can be more accurate for complex datasets given a priori knowledge of model derivatives. However, the derivatives must be known beforehand and are not often available for process data. This tool automatically predicts the gradients for a training dataset in a form usable by common GENN trainers, such as Surrogate Modeling Toolbox.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the copyright and license terms described in the LICENSE.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by bpaul4** [comment]
@lbianchi-lbl would it be useful or convenient to add testing, even if that is just a smoke test? I'm also fine overlooking code coverage for this script if that is the only issue.

**Comment by bpaul4** [comment]
@lbianchi-lbl @ksbeattie would be you available to review the methods and documentation? The documentation contains a small example script to call the method and test it. I did not add a smoke test for now, as the example tests are not part of the CI suite.

**Review by ksbeattie — APPROVED** [review]
Pretty much a rubber stamp here...

**Review by lbianchi-lbl — APPROVED** [review]
- The documentation looks thorough (although I didn't review it in depth for accuracy)
- Similarly for the code; we should figure out if or how to test scripts in the `examples` directory, but that's a more general discussion that goes beyond this PR
